src/test/coins_tests.cpp

```diff
@@ -98,6 +98,78 @@ class CCoinsViewCacheTest : public CCoinsViewCache
 
 BOOST_FIXTURE_TEST_SUITE(coins_tests, BasicTestingSetup)
 
+// This trivial contract must be tested, because instances of CCoinsView are used in production code
+BOOST_AUTO_TEST_CASE(coins_CCoinsViewContract)
+{
+    CCoinsView view;
+
+    Coin coin;
+    COutPoint const outPoint;
+    BOOST_CHECK_EQUAL(view.GetCoin(outPoint, coin), false);
+    BOOST_CHECK_EQUAL(view.GetBestBlock(), uint256());
+    BOOST_CHECK_EQUAL(view.GetHeadBlocks().size(), 0);
+
+    CCoinsMap map;
+    uint256 hashBlock;
+    BOOST_CHECK_EQUAL(view.BatchWrite(map, hashBlock), false);
+    BOOST_CHECK(view.Cursor() == nullptr);
+    BOOST_CHECK_EQUAL(view.HaveCoin(outPoint), false);
+    BOOST_CHECK_EQUAL(view.HaveCoin(outPoint), view.GetCoin(outPoint, coin));
+}
+
+// Test the contract in order to detect unintended modifications
+BOOST_AUTO_TEST_CASE(coins_CCoinsViewBackendContract)
+{
+    // CCoinsViewBacked effectively delegates to an instance of CCoinsView.
+    CCoinsView view;
+    CCoinsViewBacked vBacked(&view);
+
+    Coin coin;
+    COutPoint outPoint;
+    BOOST_CHECK_EQUAL(vBacked.GetCoin(outPoint, coin), false);
+    BOOST_CHECK_EQUAL(vBacked.HaveCoin(outPoint), false);
+    BOOST_CHECK_EQUAL(vBacked.GetBestBlock(), uint256());
+    BOOST_CHECK_EQUAL(vBacked.GetHeadBlocks().size(), 0);
+
+    CCoinsMap map;
+    uint256 hashBlock;
+    BOOST_CHECK_EQUAL(vBacked.BatchWrite(map, hashBlock), false);
+    BOOST_CHECK(vBacked.Cursor() == nullptr);
+    BOOST_CHECK_EQUAL(vBacked.EstimateSize(), 0);
+
+    // Provide an alternaitve implementation of CCoinsView with the opposite behavior
```

Comment:
Should be "alternative" :-)

Reply:
Done. Unfortunately, the build environment broke down.